text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1251 @@
+        {
+          name: 'test-public-subnet',
+          subnetType: ec2.SubnetType.PUBLIC,
+          cidrMask: SUBNET_CIDR_MASK,
+        },
+        {
+          name: 'test-private-subnet',
+          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+          cidrMask: SUBNET_CIDR_MASK,
+        }
+      ],
+      enableDnsHostnames: true,
+      enableDnsSupport: true,
+      restrictDefaultSecurityGroup: false,
+    });
+  }
+}
+```
+
+CDK team provides a comprehensive migration guide. To upgrade to `VpcV2` construct, first update your
+CDK code following the migration guide:
+
+```ts
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as ec2alpha from '@aws-cdk/aws-ec2-alpha';
+
+export class DemoStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // Create VPC with primary CIDR block
+    this.vpc = new ec2alpha.VpcV2(this, 'vpc', {
+      primaryAddressBlock: ec2alpha.IpAddresses.ipv4(`10.0.0.0/16`),
+      enableDnsHostnames: true,
+      enableDnsSupport: true,
+    });
+
+    const internetgw = new ec2alpha.InternetGateway(this, 'igw', {
+      vpc: this.vpc,
+    });
+
+    const privateSN = new ec2alpha.SubnetV2(this, PRIVATE_SUBNET_NAME, {
+      vpc: this.vpc,
+      availabilityZone: 'us-west-2',
+      ipv4CidrBlock: new ec2alpha.IpCidr(`10.0.1.0/24`),
+      subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+      mapPublicIpOnLaunch: false,
+      subnetName: PRIVATE_SUBNET_NAME,
+    });
+    cdk.Tags.of(privateSN).add('Name', `${this.stackName}/vpc/private-subnet-test-cyberark1Subnet1`);
+
+    const publicSN = new ec2alpha.SubnetV2(this, PUBLIC_SUBNET_NAME, {
+      vpc: this.vpc,
+      availabilityZone: AZ,
+      ipv4CidrBlock: new ec2alpha.IpCidr(`10.0.0.0/24`),
+      subnetType: ec2.SubnetType.PUBLIC,
+      mapPublicIpOnLaunch: true,
+      subnetName: PUBLIC_SUBNET_NAME,
+    });
+    cdk.Tags.of(publicSN).add('Name', `${this.stackName}/vpc/public-subnet-test-cyberark1Subnet1`);
+
+    new ec2alpha.Route(this, 'private-route', {
+      routeTable: privateSN.routeTable,
+      destination: "0.0.0.0/0",
+      target: new ec2alpha.RouteTargetType({ gateway: ntgwy })
+    });
+    cdk.Tags.of(privateSN.routeTable).add('Name', `${this.stackName}/vpc/private-subnet-test-cyberark1Subnet1`);
+
+    new ec2alpha.Route(this, 'public-route', {
+      routeTable: publicSN.routeTable,
+      destination: "0.0.0.0/0",
+      target: new ec2alpha.RouteTargetType({ gateway: internetgw })
+    });
+    cdk.Tags.of(publicSN.routeTable).add('Name', `${this.stackName}/vpc/public-subnet-test-cyberark1Subnet1`);
+  }
+}
+```
+
+Now, validate the construct upgrade using the CDK CLI:
+
+> Note that the flag `--unstable=construct-upgrade` is necessary to indicate that
+> it's an experiemntal feature that users are using until feature stablized.
+
+```sh
+cdk construct-upgrade --unstable=construct-upgrade --target @aws-cdk.aws-ec2-alpha.VpcV2
```

Comment:
Its somewhat confusing to use `@aws-cdk.aws-ec2-alpha.VpcV2` as the target here because the code change actually introduces many more new concstructs:

- ec2alpha.Route
- ec2alpha.SubnetV2
- ...

Can we think of a different value here that more accurately describes the migration? Maybe:

```suggestion
cdk construct-upgrade --unstable=construct-upgrade --target @aws-cdk.aws-ec2-alpha
```

?